tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,612 @@
+    {
+        "id": "123456789012345",
+        "name": "kitchen",
+        "type": "AirPlay",
+        "selected": True,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 50,
+    },
+    {
+        "id": "0",
+        "name": "Computer",
+        "type": "ALSA",
+        "selected": True,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 19,
+    },
+    {
+        "id": "100",
+        "name": "daapd-fifo",
+        "type": "fifo",
+        "selected": False,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 0,
+    },
+)
+
+
+SAMPLE_OUTPUTS_UNSELECTED = [
+    {
+        "id": "123456789012345",
+        "name": "kitchen",
+        "type": "AirPlay",
+        "selected": False,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 0,
+    },
+    {
+        "id": "0",
+        "name": "Computer",
+        "type": "ALSA",
+        "selected": False,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 19,
+    },
+    {
+        "id": "100",
+        "name": "daapd-fifo",
+        "type": "fifo",
+        "selected": False,
+        "has_password": False,
+        "requires_auth": False,
+        "needs_auth_key": False,
+        "volume": 0,
+    },
+]
+
+
+@pytest.fixture(name="config_entry")
+def config_entry_fixture():
+    """Create hass config_entry fixture."""
+    data = {
+        CONF_HOST: "192.168.1.1",
+        CONF_PORT: "2345",
+        CONF_PASSWORD: "",
+        CONF_DEFAULT_VOLUME: 0.5,
+        CONF_NAME: "Test Server",
+    }
+    return MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="",
+        data=data,
+        options={},
+        system_options={},
+        source=SOURCE_USER,
+        connection_class=CONN_CLASS_LOCAL_PUSH,
+        unique_id=CONFIG_FLOW_UNIQUE_ID,
+        entry_id=1,
+    )
+
+
+@pytest.fixture(name="get_request_return_values")
+async def get_request_return_values_fixture():
+    """Get request return values we can change later."""
+    return {
+        "config": SAMPLE_CONFIG,
+        "outputs": SAMPLE_OUTPUTS_ON,
+        "player": SAMPLE_PLAYER_PAUSED,
+        "queue": SAMPLE_TTS_QUEUE,
+    }
+
+
+@pytest.fixture(name="master")
+async def master_fixture(hass, config_entry, get_request_return_values):
+    """Create master device fixture."""
+
+    async def get_request_side_effect(update_type):
+        if update_type == "outputs":
+            return {"outputs": get_request_return_values["outputs"]}
+        return get_request_return_values[update_type]
+
+    async def add_to_queue_side_effect(uris, playback, playback_from_position=None):
+        await master._updater._update(["queue", "player"])  # sorry for touching private
```

Comment:
Patch `start_websocket_handler` instead. Then we can check the call arguments of that call and get access to the `updater._update` method. Then we can use that method to trigger updates.